runtime/onert/backend/cpu/ops/ExpandDimsLayer.cc

```diff
@@ -38,8 +38,12 @@ void ExpandDimsLayer::configure(const IPortableTensor *input, IPortableTensor *o
 
 void ExpandDimsLayer::run()
 {
-  size_t count = _input->total_size();
-  memcpy(_output->buffer(), _input->buffer(), count);
+  // output buffer equals to input buffer means that copy is not needed
+  if (_output->buffer() != _input->buffer())
```

Comment:
I have a question.
Did you have a case that input and output buffer are same?
For memcpy(), src and dest memory area must not overlap, otherwise, its behavior is undefined.

Reply:
Note that this change is a part of bigger issue implemented in https://github.com/Samsung/ONE/pull/14057. In this draft I am changing tensor allocation mechanism to share memory memory between input and output tensors of Reshape/ExpandDims/Squeeze. In other words different tensors can indicate the same buffers (it requires additional handing of such memory management).

Reply:
I've followed the draft and issue and understood.
Thanks for the explanation :)